File: backend/Mover.Entities/POCOEntities/ConductorAutorizado.cs
namespace Mover.Entities.POCOEntities
{
    public class ConductorAutorizado
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public User? User { get; set; }
        public string Nombre { get; set; } = string.Empty;
        public string Apellido { get; set; } = string.Empty;
        public string Direccion { get; set; } = string.Empty;
        public string Correo { get; set; } = string.Empty;
        public string Identificacion { get; set; } = string.Empty;
        public int Estado { get; set; }
    }
}

File: frontend/lib/domain/entities/conductor_autorizado.dart
class ConductorAutorizado {
  final int? id;
  final int? userId;
  final String nombre;
  final String apellido;
  final String direccion;
  final String correo;
  final String identificacion;
  final int? estadoConductorAutorizadoId;

  ConductorAutorizado({
    this.id,
    this.userId,
    this.nombre = "",
    this.apellido = "",
    this.direccion = "",
    this.correo = "",
    this.identificacion = "",
    this.estadoConductorAutorizadoId,
  });

  /// Método para crear un objeto `ConductorAutorizado` desde un JSON.
  factory ConductorAutorizado.fromJson(Map<String, dynamic> json) {
    return ConductorAutorizado(
      id: json['id'],
      userId: json['userId'],
      nombre: json['nombre'] ?? "",
      apellido: json['apellido'] ?? "",
      direccion: json['direccion'] ?? "",
      correo: json['correo'] ?? "",
      identificacion: json['identificacion'] ?? "",
      estadoConductorAutorizadoId: json['estadoConductorAutorizadoId'],
    );
  }

  /// Método para convertir un objeto `ConductorAutorizado` a JSON.
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'userId': userId,
      'nombre': nombre,
      'apellido': apellido,
      'direccion': direccion,
      'correo': correo,
      'identificacion': identificacion,
      'estadoConductorAutorizadoId': estadoConductorAutorizadoId,
    };
  }
}
